### PR TITLE
Update 3.4 & 3.5 changelog.

### LINKED ISSUE #16891: Upgrade go to latest patch release
### What would you like to be added?

A new go patch release has been published:

- go1.21.4 (released 2023-11-07) includes security fixes to the path/filepath package, as well as bug fixes to the linker, the runtime, the compiler, and the go/types, net/http, and runtime/cgo packages. See the [Go 1.21.4 milestone](https://github.com/golang/go/issues?q=milestone%3AGo1.21.4+label%3ACherryPickApproved) on our issue tracker for details.

We should update to this version by bumping the version in `.go-version` file in the `main` branch.

Once the changes have been merged in `main` we should also raise pull requests for `release-3.5` and `release-3.4` branches as well to update those branches to `1.20.11` so the fixes can make it into our next patch releases.

### Completion tracking

- [x] `main` - https://github.com/etcd-io/etcd/pull/16895
- [x] `release-3.5` https://github.com/etcd-io/etcd/pull/16915
- [x] `release-3.4` https://github.com/etcd-io/etcd/pull/16916
- [x] `etcd-io/bbolt` `master` https://github.com/etcd-io/bbolt/pull/607
- [x] `etcd-io/raft` `main` https://github.com/etcd-io/raft/pull/109

### Why is this needed?

Keep etcd projects up to date with the latest patch releases and incorporate the latest security updates.

### PR DESCRIPTION
**Pull request by arjunmalhotra1**
[Fix #16891(Comment)](https://github.com/etcd-io/etcd/issues/16891)

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Review by jmhbnz — APPROVED** [review]
LGTM - Thanks @arjunmalhotra1 

/lgtm
/approve

File: CHANGELOG/CHANGELOG-3.5.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ Previous change logs can be found at [CHANGELOG-3.4](https://github.com/etcd-io/
 - Fix [panic in etcd validate secure endpoints](https://github.com/etcd-io/etcd/pull/16565)
 
 ### Dependencies
-- Compile binaries using [go 1.20.10](https://github.com/etcd-io/etcd/pull/16745).
+- Compile binaries using [go 1.20.11](https://github.com/etcd-io/etcd/pull/16915).
```
Comment (ahrtr) [inline CHANGELOG/CHANGELOG-3.5.md:32]: 3.5.10 has already been released. Please create a new section for 3.5.11 to hold this change.

Comment (jmhbnz) [reply CHANGELOG/CHANGELOG-3.5.md:32]: Good catch!

Comment (arjunmalhotra1) [reply CHANGELOG/CHANGELOG-3.5.md:32]: Thank you @ahrtr. Do I only add the dependencies part in the new section for 3.5.11, and leave the other parts vacant?

Comment (jmhbnz) [reply CHANGELOG/CHANGELOG-3.5.md:32]: Example pr here for setting up changelog with new sections: https://github.com/etcd-io/etcd/pull/15841.

You only need to add the dependencies section initially, other sections will be added later as required.

Comment (ahrtr) [reply CHANGELOG/CHANGELOG-3.5.md:32]: The new section [3.5.11](https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.5.md#v3511-tbd) has already been created in another PR, please rebase this PR, and add the item under the new section. Thanks.

Comment (arjunmalhotra1) [reply CHANGELOG/CHANGELOG-3.5.md]: Thanks @jmhbnz & @ahrtr I have updated the PR. Let me know if it looks good now.

**Review by ahrtr — APPROVED** [review]
lgtm

Thanks.

**Pull request merged**